### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by Thoth1111**
In this branch, we were able to achieve the following: 
✅ install `CanCanCan` gem .
✅ Add `role` column to `users` table using a migration.
✅ Add a delete button for posts and implement functionality.
✅ Implement authorization such that the delete post button is only accessible to its author and users with admin role.
✅ Add a delete button for comments listed in a post and implement functionality.
✅ Implement authorization such that the delete comment button is only accessible to its author and users with admin role.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@ class User < ApplicationRecord
   has_many :comments, foreign_key: :author_id, dependent: :destroy
   has_many :likes, foreign_key: :author_id, dependent: :destroy
 
+  def is?(role)
```
Comment (brytebee) [inline app/models/user.rb:13]: 👏👏👏

File: app/controllers/posts_controller.rb
Comment (brytebee) [inline app/controllers/posts_controller.rb:1]: As required by t[he documentation](https://github.com/CanCanCommunity/cancancan#controller-helpers), kindly add the code below here.

```
load_and_authorize_resource
```

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Thoth1111 & @stevenmukama 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB setup.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Thoth1111) [reply app/models/user.rb:13]: Hi @brytebee , thank you for the review. I'm a bit confused with this. Is there anything I need to change or fix with this `is?` action?

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @TEAM       , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯
- [x] Blog app  running as intended💯 
 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Comment by Thoth1111** [comment]
Thank you.

**Pull request merged**